lib/ssl/src/ssl_handshake.erl

```diff
@@ -2209,24 +2203,24 @@ bad_key(#{algorithm := ecdsa}) ->
     unacceptable_ecdsa_key.
 
 cert_status_check(_,
-                  #{ocsp_state := #{ocsp_stapling := true,
-                                    ocsp_expect := stapled}},
+                  #{stapling_state := #{configured := true,
+                                        status := stapled}},
```

Comment:
`stapled` is not precise enough for TLS1.3

Reply:
Yes TLS-1.3 is handled here. Please elaborate!

Reply:
I mean in TLS1.3 maybe we should be more specific when describing for which cert we received OCSP response:
1. received_peer
2. received_chain
3. received some other OCSP response for certificate from chain
4. received OCSP response not related to certificate chain

- maybe this is indirectly handled later when cert chain gets verified, but I would like to check it
- also crl_check option by default (set to `true`) tries to verify the whole chain, then it can be set to less strict with `peer` value ... for TLS1.3 stapling we could find some similarity and maybe we should try to have it?

Reply:
maybe `received some OCSP response` is good enough at this stage ... I need to analyze it

Reply:
I do not see how nr 4 could happen!?

Reply:
The other alternatives I read as peer was stapled all certs in chain where stapled only some certs in the chain where stapled. As I said the code now would check all staples that are sent but it will not verify which certs that are stapled.
I think nr 1 and 2 are the most likly senarios.

Reply:
> I do not see how nr 4 could happen!?

bad actor knows server private key, tries to setup fake server using stolen certificate. can't provide real OCSP response and attaches a dummy one hoping it will not be checked ... a bit theoretical but still one possible flow

Reply:
Was thinking should we call it received_staple as the response in this case is the staple which I think is less obvious if we only call it response.

Reply:
let's leave for now and decide once I push updated branch. I have no strong opinion,
- `received_response` status key matches with `response` key containing CertStatus ...

Reply:
Your suggestion make sense. Response is very generic, let's use staple in this context. I've renamed both keys:
- received_response -> received_staple
- response -> staple

f06f636277